core/src/test/resources/json-ld.org/fromRdf-0020-out.jsonld

```diff
@@ -1,9 +1,7 @@
 [
   {
     "@id": "http://example.com/Subj1",
-    "http://www.w3.org/1999/02/22-rdf-syntax-ns#type" : [{
-      "@id": "http://example.com/Type"
-    }],
+    "@type" : ["http://example.com/Type"],
     "http://example.com/prop1": [{"@id": "http://example.com/Obj1"}],
```

Comment:
... while your code seems to be good in this case - compare
 [without @type](https://json-ld.org/playground/1.0/#startTab=tab-nquads&json-ld=%5B%7B%22%40id%22%3A%22http%3A%2F%2Fexample.com%2FSubj1%22%2C%22http%3A%2F%2Fwww.w3.org%2F1999%2F02%2F22-rdf-syntax-ns%23type%22%3A%5B%7B%22%40id%22%3A%22http%3A%2F%2Fexample.com%2FType%22%7D%5D%2C%22http%3A%2F%2Fexample.com%2Fprop1%22%3A%5B%7B%22%40id%22%3A%22http%3A%2F%2Fexample.com%2FObj1%22%7D%5D%2C%22http%3A%2F%2Fexample.com%2Fprop2%22%3A%5B%7B%22%40value%22%3A%22Plain%22%7D%2C%7B%22%40value%22%3A%222012-05-12%22%2C%22%40type%22%3A%22http%3A%2F%2Fwww.w3.org%2F2001%2FXMLSchema%23date%22%7D%2C%7B%22%40value%22%3A%22English%22%2C%22%40language%22%3A%22en%22%7D%5D%7D%2C%7B%22%40id%22%3A%22http%3A%2F%2Fexample.com%2FType%22%2C%22http%3A%2F%2Fwww.w3.org%2F1999%2F02%2F22-rdf-syntax-ns%23label%22%3A%5B%7B%22%40value%22%3A%22myLabel%22%7D%5D%2C%22http%3A%2F%2Fexample.com%2Fprop2%22%3A%5B%7B%22%40value%22%3A%222012-05-12%22%2C%22%40type%22%3A%22http%3A%2F%2Fwww.w3.org%2F2001%2FXMLSchema%23date%22%7D%5D%7D%5D) and [with @type](https://json-ld.org/playground/1.0/#startTab=tab-nquads&json-ld=%5B%7B%22%40id%22%3A%22http%3A%2F%2Fexample.com%2FSubj1%22%2C%22%40type%22%3A%5B%22http%3A%2F%2Fexample.com%2FType%22%5D%2C%22http%3A%2F%2Fexample.com%2Fprop1%22%3A%5B%7B%22%40id%22%3A%22http%3A%2F%2Fexample.com%2FObj1%22%7D%5D%2C%22http%3A%2F%2Fexample.com%2Fprop2%22%3A%5B%7B%22%40value%22%3A%22Plain%22%7D%2C%7B%22%40value%22%3A%222012-05-12%22%2C%22%40type%22%3A%22http%3A%2F%2Fwww.w3.org%2F2001%2FXMLSchema%23date%22%7D%2C%7B%22%40value%22%3A%22English%22%2C%22%40language%22%3A%22en%22%7D%5D%7D%2C%7B%22%40id%22%3A%22http%3A%2F%2Fexample.com%2FType%22%2C%22http%3A%2F%2Fwww.w3.org%2F1999%2F02%2F22-rdf-syntax-ns%23label%22%3A%5B%7B%22%40value%22%3A%22myLabel%22%7D%5D%2C%22http%3A%2F%2Fexample.com%2Fprop2%22%3A%5B%7B%22%40value%22%3A%222012-05-12%22%2C%22%40type%22%3A%22http%3A%2F%2Fwww.w3.org%2F2001%2FXMLSchema%23date%22%7D%5D%7D%5D&frame=%7B%22%40context%22%3A%7B%22rdf%22%3A%22http%3A%2F%2Fwww.w3.org%2F1999%2F02%2F22-rdf-syntax-ns%23%22%7D%2C%22%40id%22%3A%22http%3A%2F%2Fexample.com%2Fmain%2Fid%22%7D&context=%7B%7D) , there is no difference in the nquads.
See also https://github.com/jsonld-java/jsonld-java/issues/242#issuecomment-886961890 where this is reported to be the expected outcome.